### PR TITLE
Message input bugfix

### LINKED ISSUE #521: input field hides and appears again very slowly
steps to reproduce:

- open a chat containing some images
- open an image
- close the image by _swiping down_
- => the input field is hidden _but_ reappears slowly in a minute or so

thanks @paulaluap for figuring that out :)

### PR DESCRIPTION
**Pull request by nayooti**
closes #521

Please read my comment to #521.

This PR contains rather a workaround than a fix.

### CODE REVIEW

File: deltachat-ios/Controller/PreviewController.swift [hunk 1]
```diff
@@ -20,3 +23,53 @@ class PreviewController: QLPreviewControllerDataSource {
...
     }
 }
+
+
+class BetterPreviewController: QLPreviewController {
+    var onDismiss: VoidFunction? 
+    var urls: [URL]
+
+    private lazy var doneButtonItem: UIBarButtonItem = {
+        let button = UIBarButtonItem(title: "Fertig", style: .done, target: self, action: #selector(doneButtonPressed(_:)))
```
Comment (cyBerta) [inline deltachat-ios/Controller/PreviewController.swift]: the title string should be translated, not hard coded

File: deltachat-ios/Controller/PreviewController.swift [hunk 2]
```diff
@@ -20,3 +23,53 @@ class PreviewController: QLPreviewControllerDataSource {
         return urls[index] as QLPreviewItem
     }
 }
+
+
+class BetterPreviewController: QLPreviewController {
```
Comment (cyBerta) [inline deltachat-ios/Controller/PreviewController.swift]: Can I suggest to rename the controller :)? Maybe MediaPreviewController or sth like that?

File: deltachat-ios/Controller/PreviewController.swift [hunk 3]
```diff
@@ -2,6 +2,9 @@ import QuickLook
 import UIKit
 
 class PreviewController: QLPreviewControllerDataSource {
```
Comment (cyBerta) [inline deltachat-ios/Controller/PreviewController.swift]: maybe this class can be renamed as well now that there is a similar named one?

**Comment by r10s** [comment]
so i think, this one can be closed as included in #560 ?

**Comment by nayooti** [comment]
Exactly. Can be closed. And #560  should be seen/considered as a temporary fix.

**Pull request closed without merging**